overlord/snapstate/snapstate.go

```diff
@@ -4146,6 +4147,125 @@ func GadgetConnections(st *state.State, deviceCtx DeviceContext) ([]gadget.Conne
 	return gadgetInfo.Connections, nil
 }
 
+// deriveNeededSnapRevisions returns revisions that need to be kept for the given snap.
+//
+// A revision is needed only if:
```

Comment:
Just to make unambiguous that a revision is kept if **any** of the conditions are met
```suggestion
// A downloaded file is only kept if any of the following are true:
```